### PR TITLE
[elisp] add elisp-def

### PR DESCRIPTION
**Pull request by thanhvg**
this package provide highly accurate jump to the definition, it even works with
local variable

### CODE REVIEW

File: layers/+lang/emacs-lisp/packages.el [hunk 1]
```diff
@@ -160,6 +161,10 @@
       (spacemacs/set-leader-keys-for-major-mode 'emacs-lisp-mode
         "cl" 'auto-compile-display-log))))
 
+(defun emacs-lisp/init-elisp-def ()
+  (use-package elisp-def
+    :defer (spacemacs/defer)))
```
Comment (lebensterben) [inline layers/+lang/emacs-lisp/packages.el:166]: normally we use :defer t
why do you use (spacemacs/defer) here?

Comment (thanhvg) [reply layers/+lang/emacs-lisp/packages.el:166]: I don't know somehow other packages in the elisp layers follow this convention. Its goal I assume is to work with dump feature which I never try.

Comment (lebensterben) [inline layers/+lang/emacs-lisp/packages.el:166]:
```suggestion
    :defer t))
```

**Comment by lebensterben** [comment]
Please update CHANGELOG.develop and I will merge this.

**Comment by lebensterben** [comment]
Merged. Thanks for your contributions.

**Pull request merged**